test/cpp/end2end/async_end2end_test.cc

```diff
@@ -858,9 +859,12 @@ class AsyncEnd2endServerTryCancelTest : public AsyncEnd2endTest {
 
     bool expected_server_cq_result = true;
     bool ignore_cq_result = false;
+    bool want_done_tag = false;
 
     if (server_try_cancel == CANCEL_BEFORE_PROCESSING) {
-      ServerTryCancel(&srv_ctx);
+      srv_ctx.TryCancel();
+      Verifier(GetParam()).Expect(11, true).Verify(cq_.get());
```

Comment:
Ok. No strong reason. Just that one might wonder why this is being treated as a special case :).
 Actually the CANCEL_*_PROCESSING states are all mutually exclusive.. So there shouldn't be any sharing of state between those cases. I don't feel strongly about and its up to you..